### PR TITLE
Added MongoDB documentation.

### PR DESCRIPTION
**Pull request by dat-a-man**
### Description

Added MongoDB documentation.

### Related Issues

- Fixes #...
- Closes #...
- Resolves #...

### CODE REVIEW

File: docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+Args:
+
+The arguments `connection_url`, `database`, `incremental` and `write_disposition` are the same as
+defined for [Source `mongodb`](#source-mongodb).
+
+`collection`: Name of the collection to load.
+
+Returns:
+
+Iterable [DltResource]: A list of DLT resources ("collection_documents") for each collection to be
```
Comment (AstrakhantsevaAA) [inline docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md:278]: it returns only one resource

Comment (AstrakhantsevaAA) [reply docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md:278]: there is a mistake https://github.com/dlt-hub/verified-sources/blob/master/sources/mongodb/__init__.py#L83C1-L83C89

File: docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+    write_disposition=write_disposition,
+)(client, collection_obj, incremental=incremental)
+```
+
+`name`: Collection object's name.
+
+Parameters `primary_key` and `write_disposition` are as mentioned in
+[resource `collection_documents`.](#resource-collectiondocuments)
+
+### Create your own pipeline
```
Comment (AstrakhantsevaAA) [inline docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md:301]: i would recommend take the existed README as an example, there is good style.
https://github.com/dlt-hub/verified-sources/blob/master/sources/mongodb/README.md#how-to-use

File: docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+    name=collection_obj.name,
+    primary_key="_id",
+    write_disposition=write_disposition,
+)(client, collection_obj, incremental=incremental)
+```
+
+`name`: Collection object's name.
+
+Parameters `primary_key` and `write_disposition` are as mentioned in
+[resource `collection_documents`.](#resource-collectiondocuments)
```
Comment (AstrakhantsevaAA) [inline docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md:299]: link doesn't work

File: docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+   load_data = mongodb(incremental=dlt.sources.incremental("date")).with_resources("collection_1")
+   load_info = pipeline.run(load_data, write_disposition = "merge")
+   print(load_info)
+   ```
+   > Data is loaded incrementally based on "date" field.
+
+1. To load data from a particular collection say "movies" incrementally:
+
+   ```python
+   load_data = mongodb_collection(
```
Comment (AstrakhantsevaAA) [inline docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md:344]: we need to explain why we used mongodb as a source above, and now we use mongodb_collection. what is the difference between them?

File: docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md (hunk 3 above)
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md:299]: fixed

File: docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md (hunk 4 above)
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md:344]: added context

File: docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md [hunk 5]
```diff
@@ -0,0 +1,384 @@
...
+| Replica Set         | Set up for high availability with data replication across multiple MongoDB instances. | "mongodb://dbuser:passwd@replica.set:27017"       |
+| Sharded Cluster     | Scalable distribution of datasets across multiple MongoDB instances.                  | "mongodb://dbuser:passwd@shard.cluster:27017"     |
+| Kubernetes          | Deploy on Kubernetes using Helm charts or operators.                                  | "mongodb://dbuser:passwd@k8s.cluster:27017"       |
+| Manual Tarball      | Install directly from the official MongoDB tarball, typically on Linux.               | "mongodb://dbuser:passwd@tarball.host:27017"      |
+
+> Note: The provided URLs are example formats; adjust as needed for your specific setup.
+
+#### Grab `database and collections`
+
+1. To grab "database and collections" you must have MongoDB shell installed.
```
[the remark this answers is not in the document]
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md:62]: added doc link.

File: docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md [hunk 6]
```diff
@@ -0,0 +1,384 @@
...
+
+1. Modify the example URLs with your credentials (dbuser & passwd) and host details.
+
+1. Connect to MongoDB
```
[the remark this answers is not in the document]
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md:66]: fixed

File: docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md [hunk 7]
```diff
@@ -0,0 +1,384 @@
...
+      show collections
+      ```
+
+1. Disconnect:
+
+   ```bash
+   exit
+   ```
+
+Please note the database and collection names for data loading.
```
[the remark this answers is not in the document]
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md:95]: added context

File: docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md [hunk 8]
```diff
@@ -0,0 +1,384 @@
...
+   file with care and keep it safe. Here's what the file looks like:
+
+   ```toml
+   # put your secret values and credentials here
+   # do not share this file and do not push it to github
+   [sources.mongodb]
+   connection_url = "mongodb connection_url" # please set me up!
+   ```
+
+1. Replace the connection_url value with the [previously copied one](#grab-connectionurl) to ensure
```
[the remark this answers is not in the document]
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md:148]: fixed.

File: docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md [hunk 9]
```diff
@@ -0,0 +1,384 @@
...
+### Prepare your data
+
+Data in MongoDB is stored in BSON (Binary JSON) format, which allows for embedded documents or
+nested data. It employs a flexible schema, and its key terms include:
+
+`Documents`: Key-value pairs representing data units.
+
+`Collections`: Groups of documents, similar to database tables but without a fixed schema.
+
+`Databases`: Containers for collections; a single MongoDB server can have multiple databases
```
[the remark this answers is not in the document]
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md:106]: done

File: docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md [hunk 10]
```diff
@@ -0,0 +1,384 @@
...
+    database: Optional[str] = dlt.config.value,
+    collection: str = dlt.config.value,
+    incremental: Optional[dlt.sources.incremental] = None,  # type: ignore[type-arg]
+    write_disposition: Optional[str] = dlt.config.value,
+) -> Any:
+```
+
+Args:
+
+The arguments `connection_url`, `database`, `incremental` and `write_disposition` are the same as
```
[the remark this answers is not in the document]
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md:271]: done

File: docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md [hunk 11]
```diff
@@ -0,0 +1,384 @@
...
+
+`write_disposition`: Writing mode: "replace", "append", or "merge".
+
+The `spec` parameter deploys 'MongoDbCollectionConfiguration' for MongoDB setup. Each resource is
+invoked with the MongoDB client, current collection, and incremental parameter, preparing a DLT
+resource for each item in collection_list.
+
+### Source `mongo_collection`
+
+This function fetches data from a MongoDB collection, returning resources for each collection to be loaded.
```
[the remark this answers is not in the document]
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md:257]: corrected.

File: docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md:271]: deleted

File: docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md (hunk 2 above)
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md:301]: deleted

File: docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md [hunk 12]
```diff
@@ -0,0 +1,384 @@
...
+
+`incremental`: Option for incremental data loading.
+
+`write_disposition`: Writing mode: "replace", "append", or "merge".
+
+Returns:
+
+`Iterable[DltResource]`: A list of DLT resources for each specified collection.
+
+### Resource `collection.name`
```
[the remark this answers is not in the document]
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md:228]: deleted

File: docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md (hunk 1 above)
Comment (dat-a-man) [reply docs/website/docs/dlt-ecosystem/verified-sources/mongodb.md:278]: corrected

**Review by adrianbr — APPROVED** [review]

**Pull request merged**